ros2topic/ros2topic/verb/echo.py

```diff
@@ -112,10 +131,9 @@ def subscriber(node, topic_name, message_type, callback):
     msg_module = import_message_type(topic_name, message_type)
 
     node.create_subscription(
-        msg_module, topic_name, callback, qos_profile_sensor_data)
+        msg_module, topic_name, callback, qos_profile)
 
-    while rclpy.ok():
-        rclpy.spin_once(node)
+    rclpy.spin(node)
```

Comment:
Any reason for removing the while loop?

Reply:
There's a good reason. creating an executor, which rclpy does, is insanely expensive. We're talking like 30 ms of cpu work for some reason.

Reply:
Part of what I was trying to do here was put some consistency between pub and echo. `spin` exists for this exact reason anyways and it's what `pub.py` does

Reply:
https://discourse.ros.org/t/singlethreadedexecutor-creates-a-high-cpu-overhead-in-ros-2/10077

Reply:
Thanks!